yt/data_objects/tests/test_particle_trajectories_pytest.py

```diff
@@ -0,0 +1,150 @@
+    ("all", "particle_velocity_z"),
+]
+
+
+@pytest.fixture
+def particle_trajectories_test_dataset():
+    n_particles = 2
+    n_steps = 2
+    ids = np.arange(n_particles, dtype="int64")
+    data = {"particle_index": ids}
+    fields = [
+        "particle_position_x",
+        "particle_position_y",
+        "particle_position_z",
+        "particle_velocity_x",  # adding a non-default field
+        "particle_index",
+    ]
+    negative = [False, False, False, True, False]
+    units = ["cm", "cm", "cm", "cm/s", "1"]
+
+    ts = DatasetSeries(
+        [
+            fake_particle_ds(
+                fields=fields,
+                negative=negative,
+                units=units,
+                npart=n_particles,
+                data=data,
+            )
+            for i in range(n_steps)
+        ]
+    )
+    return ts
+
+
+def test_uniqueness():
+    n_particles = 2
+    n_steps = 2
+    ids = np.arange(n_particles, dtype="int64") % (n_particles // 2)
+    data = {"particle_index": ids}
+    fields = [
+        "particle_position_x",
+        "particle_position_y",
+        "particle_position_z",
+        "particle_index",
+    ]
+    negative = [False, False, False, False]
+    units = ["cm", "cm", "cm", "1"]
+
+    ts = DatasetSeries(
+        [
+            fake_particle_ds(
+                fields=fields,
+                negative=negative,
+                units=units,
+                npart=n_particles,
+                data=data,
+            )
+            for i in range(n_steps)
+        ]
+    )
+
+    assert_raises(YTIllDefinedParticleData, ts.particle_trajectories, [0])
+
+
+def test_ptype():
+    n_particles = 100
+    fields = [
+        "particle_position_x",
+        "particle_position_y",
+        "particle_position_z",
+        "particle_index",
+        "particle_dummy",
+    ]
+    negative = [False, False, False, False, False]
+    units = ["cm", "cm", "cm", "1", "1"]
+
+    # Setup filters on the 'particle_dummy' field, keeping only the first 50
+    @particle_filter(name="dummy", requires=["particle_dummy"])
+    def dummy(pfilter, data):
+        return data[(pfilter.filtered_type, "particle_dummy")] <= n_particles // 2
+
+    # Setup fake particle datasets with repeated ids. This should work because
+    # the ids are unique among `dummy_particles` so let's test this
+    data = {
+        "particle_index": np.arange(n_particles) % (n_particles // 2),
+        "particle_dummy": np.arange(n_particles),
+    }
+    all_ds = [
+        fake_particle_ds(
+            fields=fields, negative=negative, units=units, npart=n_particles, data=data
+        )
+    ]
+    for ds in all_ds:
+        ds.add_particle_filter("dummy")
+    ts = DatasetSeries(all_ds)
+
+    # Select all dummy particles
+    print(ts[0].derived_field_list)
+    ids = ts[0].all_data()["dummy", "particle_index"]
+
+    # Build trajectories
+    ts.particle_trajectories(ids, ptype="dummy")
+
+
+@pytest.mark.parametrize("ptype", [None, ("io")])
```

Comment:
The use of parentheses seems to in indicate you meant to use a tuple here, I think ? The issue is that it's the not parentheses that make a tuple, it's really the comma ! in other words, `("io") == "io"`.
```suggestion
@pytest.mark.parametrize("ptype", [None, "io"])
```